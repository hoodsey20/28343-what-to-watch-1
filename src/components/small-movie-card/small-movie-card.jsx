import React from 'react';
import PropTypes from 'prop-types';

import VideoPlayer from '../video-player/video-player.jsx';

const SmallMoovieCard = ({movie, clickHandler, hoverHandler, leaveHandler, isPlaying}) => {
  const {name, cover, src, id} = movie;

  return (
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={() => hoverHandler(id)}
      onMouseLeave={leaveHandler}
    >
      <div className="small-movie-card__image">
        <VideoPlayer
          cover={cover}
          src={src}
          isPlaying={isPlaying}
        />
      </div>
      <h3
        className="small-movie-card__title"
        onClick={(evt) => {
          evt.preventDefault();
          clickHandler(movie);
        }}
      >
        <a className="small-movie-card__link" href="movie-page.html">{name}</a>
      </h3>
    </article>

  );
};

SmallMoovieCard.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    cover: PropTypes.string,
  }).isRequired,
  isPlaying: PropTypes.bool.isRequired,
  clickHandler: PropTypes.func,
  hoverHandler: PropTypes.func,
  leaveHandler: PropTypes.func,
};

export default SmallMoovieCard;
