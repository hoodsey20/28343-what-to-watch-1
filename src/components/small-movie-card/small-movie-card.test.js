import React from 'react';
import renderer from 'react-test-renderer';

import {SmallMoovieCard} from './small-movie-card.jsx';

const mockMovie = {
  id: 1,
  name: `Fantastic Beasts: The Crimes of Grindelwald`,
  preview_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  src: [
    `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  ]
};

it(`SmallMoovieCard card is rendered correctly`, () => {

  const tree = renderer.create(<SmallMoovieCard
    movie={mockMovie}
    onClick={jest.fn()}
    isPlaying={false}
    renderVideo={() => <div />}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
