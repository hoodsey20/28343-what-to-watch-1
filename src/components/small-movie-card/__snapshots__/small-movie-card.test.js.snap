// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallMoovieCard card is rendered correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      muted={true}
      poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      width="250"
    >
      <source
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        type="video/mp4"
      />
      <source
        src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        type="video/webm"
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;
